app/controllers/application_controller.rb

```diff
@@ -121,4 +121,18 @@ def present_new_navigation?
     end
   end
   helper_method :present_new_navigation?
+
+private
+
+  def default_url_options
+    token.merge(cache)
```

Comment:
this is a little confusing at first glance. How would you feel about
```
{}
  .merge(token)
  .merge(cache)
```

or similar?